src/MatlabOutput.h

```diff
@@ -0,0 +1,19 @@
+#ifndef HALIDE_MATLAB_OUTPUT_H
+#define HALIDE_MATLAB_OUTPUT_H
+
+/** \file
+ *
+ * Provides an output function to generate a Matlab mex API compatible object file.
+ */
+
+#include "Module.h"
+
+namespace Halide {
+
+/** Compile a module to an object file suitable for use with Matlab's mex feature. */
```

Comment:
Part of the reason this is vague is I'm not exactly sure yet. I have something I do to test it right now, but I'd like to build a wrapper and have people use that instead. Right now it's a few steps to use it (build the object, run mex, call the function), but there's no reason for that to be more than one step.

I'll add some comments about the mexFunction it generates, and leave the rest to the wrapper itself.